### PR TITLE
Revert "Turning off user agent changes temporarily"

### PR DESCRIPTION
**Pull request by thomasmaclean**
Reverts GoogleCloudPlatform/gsutil#1017

Turning on new user agent fields.

### CODE REVIEW

File: gslib/utils/user_agent_helper.py [hunk 1]
```diff
@@ -39,19 +40,27 @@ def GetUserAgent(args, metrics_off=True):
...
-      cloud_uri_pattern = re.compile(r'^(gs|s3)\://')
-      cloud_uris = [arg for arg in args if cloud_uri_pattern.search(arg)]
-      cloud_uri_dst = cloud_uri_pattern.search(args[-1])
+      # Any cp, mv or rsync commands that use daisy chain mode should be noted
+      # as that represents a unique use case that may be better served by the
+      # storage transfer service.
+      storage_urls = []
+      for arg in args:
+        try:
+          storage_urls.append(StorageUrlFromString(arg))
```
Comment (dilipped) [inline gslib/utils/user_agent_helper.py:49]: Can this handle all the flags correctly?

File: gslib/utils/user_agent_helper.py [hunk 2]
```diff
@@ -32,11 +33,28 @@ def GetUserAgent(args, metrics_off=True):
...
+
+  if len(args) > 0:
+    user_agent += ' command/%s' % args[0]
+
+    if args[0] in ['cp', 'mv', 'rsync']:
+      # Any cp, mv or rsync commands that use daisy chain mode should be noted
+      # as that represents a unique use case that may be better served by the
+      # storage transfer service.
+      try:
+        src = StorageUrlFromString(args[-2])
```
Comment (dilipped) [inline gslib/utils/user_agent_helper.py:47]: Potential IndexError. Maybe you will need a test case for ['cp'] like you have one for help

File: gslib/utils/user_agent_helper.py [hunk 3]
```diff
@@ -32,11 +33,28 @@ def GetUserAgent(args, metrics_off=True):
...
+          user_agent += '-DaisyChain'
+      except InvalidUrlError:
+        pass
 
   if system_util.InvokedViaCloudSdk():
     user_agent += ' google-cloud-sdk'
     if system_util.CloudSdkVersion():
       user_agent += '/%s' % system_util.CloudSdkVersion()
 
+  print(user_agent)
```
Comment (dilipped) [inline gslib/utils/user_agent_helper.py:59]: This should be removed?

File: gslib/utils/user_agent_helper.py (hunk 2 above)
Comment (thomasmaclean) [reply gslib/utils/user_agent_helper.py:47]: Good call!

**Pull request merged**